packages/golden-layout/src/LayoutManager.ts

```diff
@@ -536,8 +566,36 @@ export class LayoutManager extends EventEmitter {
     }
 
     config.id = config.id ?? getUniqueId();
-    contentItem = new this._typeToItem[config.type](this, config, parent);
-    return contentItem;
+
+    if (config.type === 'stack') {
+      if (parent == null) {
+        throw new Error(
+          'Stacks can only be created as children of a row or column'
+        );
+      }
+
+      // The type narrowing isn't able to infer `parent` if of type `RowOrColumn`,
+      // but the type signature of `createContentItem` should enforce this.
+      return new Stack(this, config, parent as RowOrColumn);
+    }
+
+    if (config.type === 'row') {
+      if (parent == null) {
+        throw new Error('Rows require a parent');
+      }
+
+      return new RowOrColumn(false, this, config, parent);
+    }
+
+    if (config.type === 'column') {
+      if (parent == null) {
+        throw new Error('Columns require a parent');
+      }
+
+      return new RowOrColumn(true, this, config, parent);
+    }
+
+    return new Component(this, config, parent);
```

Comment:
@mofojed e2e tests fail for `react-component` type if I require `parent` to be defined, so I tweaked the `Component` config type internally. The others (RowOrColumn and Stack) seem to require a parent as far as I can tell. I'm guessing a bit here, so if you have any more concrete insight into which cases would always require a parent vs not, that would be helpful

Reply:
Unit tests don't seem to like requiring parents for rows. It's possible that all of these items should have parent marked as optional and need some runtime refactoring internally, but I'm not sure

Reply:
I ended up updating the content items to allow `null` parent instead since it seems that is what can happen at runtime.